Changed copyright to 2019

tezc opened the pull request.

mmedenjak commented:
originally by @tkountis:

> this might help when in diff view
> ```
> var pendingDiffs = document.getElementsByClassName('load-diff-button');
> for (var i = 0; i < pendingDiffs.length; i++) {
>    pendingDiffs[i].click();
> }
> ```
Careful as this may blow up your browser's JS engine :)

blazember commented:
I leave my comments as general ones, since the `files changed` tab kills my browser :)

I used `git show --numstat | grep -v "^\s*1\s*1\s*.*$"` to find files where the number of the line changes indicate more than a simple `2018` -> `2019` update.

Files have double copyright banner header, think the second should be removed:
- `AddMultiMapConfigMessageTask.java`
- `AddQueueConfigMessageTask.java`
- `BinaryInterface.java`

Banner added to these files, this looks fine:
- `hazelcast/src/main/resources/META-INF/services/com.hazelcast.DataSerializerHook`
- `hazelcast/src/main/resources/META-INF/services/com.hazelcast.instance.NodeExtension`
- `hazelcast/src/main/resources/META-INF/services/com.hazelcast.spi.discovery.DiscoveryStrategyFactory`
- `hazelcast/src/main/resources/META-INF/services/javax.cache.spi.CachingProvider`

Also used `git show | grep "^-\s.*" | grep -v "2018"` to find changes where we modify years from other than `2018`. Based on this, the files below need to take a look.

Changes from `2015`:
- `BiInt2ObjectMap.java` 
- `Hashing.java`
- `Int2ObjectHashMap.java`
- `IntHashSet.java`
- `IntIterator.java`
- `Long2LongHashMap.java`
- `Long2ObjectHashMap.java`
- `LongHashSet.java`
- `LongIterator.java`
- `MapDelegatingSet.java`
- `BiInt2ObjectMapTest.java`
- `Int2ObjectHashMapTest.java`
- `IntHashSetTest.java`
- `Long2LongHashMapTest.java`
- `Long2ObjectHashMapTest.java`
- `LongHashSetTest.java`

Change from `2017`:
- `ThreadLocalLeakTestUtils.java`

All changes are like this:
```
- * Modified work Copyright (c) 201(5|7), Hazelcast, Inc. All Rights Reserved.
+ * Modified work Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
```
I _feel_ this should be 
```
+ * Modified work Copyright (c) 201(5|7)-2019, Hazelcast, Inc. All Rights Reserved.
```
since until that year the original author had the copyright. Guess it's not correct to overlap in the copyright.

tezc commented:
Thanks for the detailed review @blazember. Addressed all comments.

Also fixed overlapping license issue for : 

- `Pipe.java`
- `QueuedPipe.java`
- `ManyToOneConcurrentArrayQueue.java`
- `OneToOneConcurrentArrayQueue.java`
- `PhiAccrualFailureDetector`
